app/src/main/java/org/simple/clinic/patient/PatientSearchResult.kt

```diff
@@ -95,7 +95,9 @@ data class PatientSearchResult(
     val phoneUpdatedAt: Instant?,
 
     @Embedded(prefix = "lastSeen_")
-    val lastSeen: LastSeen?
+    val lastSeen: LastSeen?,
+
+    val isAtAssignedFacility: Boolean?
```

Comment:
Is it necessary to add this as a property in the search result view? I thought we would only need it in the SQL query to sort the results, right?

Reply:
It would return the sorted results but we still need to decide where to place the header right? currently, there is no indicator to decide headers in between the assigned and non-assigned facility patients. Do you think there is some other way we can achieve this?

Reply:
This is how I am planning to add the headers.

<img width="857" alt="Screenshot 2020-12-28 150559" src="https://user-images.githubusercontent.com/6140516/103204841-39a6a080-491e-11eb-9a99-263ea94d14a3.png">

Reply:
Ah, I see. I was thinking we could add the patient assigned facility ID separately to the `PatientSearchResult` type.

Reply:
I am unsure how that's going to work out. If we pass the `assignedFacilityId` via `PatientSearchResult`, we will still need to have a conditional to verify whether or not the current facility id matches the assigned facility id and then place the headers. I thought having a boolean value would better in this scenario since the data is already sorted at SQL level.

Reply:
The issue that I am talking about here is irrelevant to whether we are sorting in SQL or not. Sorting and grouping here are runtime operations since we will not know whether a patient is assigned to the given facility or not until we run the query runtime. My problem is adding the `isAssignedToCurrentFacility` boolean property to the `PatientSearchResult` type.

When it comes to a database view, it cannot have runtime properties since the SQL statement is a part of the view declaration. So something like "is the patient assigned to the current facility" should not be tracked in the view since the view itself cannot tell what the current facility is. However, something like "what is the patient's assigned facility" is something that can be tracked since it is a static property tied to the patient.

What we _should_ be doing here is tracking each patient's assigned facility in the search results and then doing a runtime check to split them into groups.

Reply:
My only concern is it doesn't matter if we are sorting them at SQL level anymore since we will be doing the sorting and partition when generating the item view types list. Is that right?

Reply:
Yeah, that's a redundant sorting for now. But it will still make the runtime sort operation more efficient since the source results are already sorted.

In any case, once we eventually move this class to a paginated fashion, the sorting at the SQL layer will be needed again, so 🤷🏼